core/services/relay/delegate.go

```diff
@@ -106,14 +106,15 @@ func (d delegate) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (types
 			return nil, errors.Wrap(err, "error on 'spec.RelayConfig' unmarshal")
 		}
 
-		programID, err := solanaGo.PublicKeyFromBase58(spec.ContractID)
+		// use state account as contract ID (unique for each feed, program account is not)
+		stateID, err := solanaGo.PublicKeyFromBase58(spec.ContractID)
```

Comment:
Can we get a test case that would have failed before?

Reply:
hmm this is fundamentally a rename/reinterpretation, not sure how you'd do that

Reply:
I had an idea yesterday, i was going to write a quick test case that just parses the job spec into the relayer. so that should confirm that the parameters passed to the relayer correspond to the correct job spec. 

it can't check the original bug like connor mentioned, but it can validate that the implementation matches the expected job spec format